### DISCUSSION TITLE
DDBoost Backup Promotion

### DISCUSSION THREAD

**Discussion by fabiopennisi** [olahallengren/sql-server-maintenance-solution#702] [Ideas]
Hey Ola (and contributors),

I've been using these great scripts for as long as I can remember, I'm really thankful for the awesome work you've done.

I've been working with Dell EMC to get Data Domain Boost (with Ola scripts) working in our environment. During testing, I noticed the `@ChangeBackupType` parameter had no effect as it would ordinarily (with other backup software). So I went digging...

I see there's a blanket rule (`SET @CurrentCommand += ' -a "BACKUP_PROMOTION=NONE"'`) that effectively "disables" backup promotion for DDBoost, which means the `@ChangeBackupType` parameter has no effect. I've traced it back to [this issue](https://github.com/olahallengren/sql-server-maintenance-solution/issues/464).

Any chance you can please adapt the [backup proc](https://github.com/olahallengren/sql-server-maintenance-solution/blob/master/DatabaseBackup.sql) to include backup promotion for DDBoost?

**Option 1:**
At the very least (and arguably the easiest approach):
- if `@ChangeBackupType = 'Y' AND @BackupSoftware = 'DATA_DOMAIN_BOOST'`
- then `SET @CurrentCommand += ' -a "BACKUP_PROMOTION=ALL"'`

**Option 2:**
If someone's up for the challenge, to support all available options:

- **ALL:** Enables backup promotion to occur in any applicable scenario.
- **NONE:** Disables all backup promotion.
- **NONE_WITH_WARNINGS:** Disables backup promotion, but logs a warning when backup promotion would normally occur.
- **SKIP_RECOVERY_MODEL:** Disables database recovery model change detection. Backup promotion as a result of recovery model change will not occur, but backup promotion in other scenarios will still occur.
- **SKIP_RECOVERY_MODEL_WITH_WARNINGS:** Enables database recovery model change detection, but if a recover model change is discovered, logs a warning instead of promoting the backup. Backup promotion in other scenarios will still occur.

Can someone please assist?

Cheers

**Comment by ozprem** [comment]
I totally agree .. This should really be an option and not a default of NONE as any log gaps do not force a full backup and render those backups useless.

**Comment by mezumaraizu** [comment]
Agree with this suggestion, would be great to have this (or code to this effect) added to the solution.

**Comment by fabiopennisi** [comment]
@olahallengren, how's it going? Any chance this can be added please?
